src/formatDate.js

```diff
@@ -8,7 +8,26 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separatorFrom = fromFormat[3];
+  const dateParts = date.split(separatorFrom);
+  const dateMap = {};
+
+  for (let i = 0; i < 3; i++) {
+    dateMap[fromFormat[i]] = dateParts[i];
+  }
+
+  if (fromFormat.includes('YYYY') && toFormat.includes('YY')) {
+    dateMap['YY'] = dateMap['YYYY'].slice(2);
+  } else if (fromFormat.includes('YY') && toFormat.includes('YYYY')) {
+    const year = parseInt(dateMap['YY']);
+
+    dateMap['YYYY'] = year < 30 ? '20' + dateMap['YY'] : '19' + dateMap['YY'];
+  }
+
+  const separatorTo = toFormat[3];
+  const newDateParts = toFormat.slice(0, 3).map((part) => dateMap[part]);
```

Comment:
The mapping of date parts to the new format is done correctly. However, it's important to consider that the 'map' function might not preserve the order of the date parts if the 'toFormat' array is not in the correct order. Ensure that the 'toFormat' array is always in the order of 'YYYY', 'MM', 'DD' when passed to this function, as the current logic assumes this order.